lib/next_ls.ex

```diff
@@ -149,6 +159,31 @@ defmodule NextLS do
      )}
   end
 
+  def handle_request(
+        %TextDocumentCodeAction{
+          params: %CodeActionParams{
+            context: %CodeActionContext{diagnostics: diagnostics},
+            text_document: %TextDocumentIdentifier{uri: uri}
+          }
+        },
+        lsp
+      ) do
+    code_actions =
+      for %Diagnostic{} = diagnostic <- diagnostics,
+      # Note: the diagnostic data doesn't have a namespace attribute yet,
+      # this is only so that there are no compile error
+      # Maybe add the namespace to the data attribute
+      # or retrieve the diagnostics for the current file from diagnostic cache
```

Comment:
Doing the latter, i.e. retrieving the diagnostic from the cache would probably be wrong, since as documented for the diagnostics field in `%CodeActionContext`, it says:
```
  * diagnostics: An array of diagnostics known on the client side overlapping the range provided to the
    `textDocument/codeAction` request. They are provided so that the server knows which
    errors are currently presented to the user for the given range. There is no guarantee
    that these accurately reflect the error state of the resource. The primary parameter
    to compute code actions is the provided range.
```
It seems to me that getting the diagnostics from the cache could lead to situations where the client isn't seeing a diagnostic but we send back a code action for it or the vice versa where the client sends a diagnostic that we ignore.